tests/infrastructure/test_dynamo_db_stack.py

```diff
@@ -0,0 +1,96 @@
+import aws_cdk as cdk
+import pytest
+from aws_cdk.assertions import Match as match
+from aws_cdk.assertions import Template
+
+from sds_data_manager.stacks.dynamodb_stack import DynamoDB
+
+
+@pytest.fixture()
+def on_demand_dynamodb(sds_id="test"):
+    app = cdk.App()
+
+    stack_name = f"on-demand-stack-{sds_id}"
+    stack = DynamoDB(
+        app,
+        stack_name,
+        sds_id,
+        table_name=f"table-{sds_id}",
+        partition_key="filename",
+        sort_key="instrument",
+        env=None,
+    )
+    template = Template.from_stack(stack)
+    return template
+
+
+@pytest.fixture()
+def provisioned_dynamodb(sds_id="test"):
+    app = cdk.App()
+
+    stack_name = f"provisioned-stack-{sds_id}"
+    stack = DynamoDB(
+        app,
+        stack_name,
+        sds_id,
+        table_name=f"table-{sds_id}",
+        partition_key="filename",
+        sort_key="instrument",
+        env=None,
+        on_demand=False,
+        write_capacity=100,
+        read_capacity=100,
+    )
+    template = Template.from_stack(stack)
+    return template
+
+
+def test_table_count(on_demand_dynamodb, provisioned_dynamodb):
```

Comment:
```suggestion
@pytest.mark.parametrize("template", [on_demand_dynamodb, provisioned_dynamodb])
def test_table_count(template):
    template.resource_count_is("AWS::DynamoDB::Table", 1)
```

I think this would work with parametrization on most of these tests (the test_billing_mode would be trickier though, so probably not worth doing).

Reply:
Yeah, billing mode test was little tricky. But it's good to know we can do that.